src/QuantumNLDiffEq.jl

```diff
@@ -1,3 +1,49 @@
 module QuantumNLDiffEq
+	import Yao: AbstractBlock, zero_state, expect, dispatch!, dispatch, chain
+	import Flux: update!, ADAM, mse
+	import Zygote: Buffer, gradient, ignore
+	
+	abstract type FeatureMap end
+	abstract type BoundaryHandling end
+	
+	struct Product <: FeatureMap end
+	struct Chebyshev <: FeatureMap end
+	struct Pinned <: BoundaryHandling end
+	struct Floating <: BoundaryHandling end
+		
+	function map_to_circuit(a, b, ::Chebyshev)
+		return a * b
+	end
 
+	function map_to_circuit(a, b, ::Product)
+		return a
+	end
+	
+	include("phi.jl")
+	include("new_circuit.jl")
+	include("calculate_diff_evalue.jl")
+	
+	loss_calc(diff_evalue, evalue, x) = mse(diff_evalue + 8*evalue*(0.1 + tan(8*x)), 0)
+	
+	function loss(feature_map_circuit::AbstractBlock, cost::AbstractBlock, var::AbstractBlock, M::AbstractVector, N::Int, u_0::Real, theta; mapping::FeatureMap)
+		Loss_diff = 0.0
+		for x in 1:length(M)
+			evalue = expect(cost, zero_state(N)=>new_circuit(feature_map_circuit, var, M[x], theta, N; mapping)) - expect(cost, zero_state(N)=>new_circuit(feature_map_circuit, var, M[1], theta, N; mapping)) + u_0
```

Comment:
Would it be ok to make the `evalue` line more DRY by extracting the `expect` terms into a separate function?

```suggestion
                _get_expect(mx) = expect(
			cost,
			zero_state(N)=>new_circuit(feature_map_circuit, var, mx, theta, N; mapping)
		)
		Loss_diff = 0.0
		for x in 1:length(M)
			evalue = _get_expect(M[x]) - _get_expect(M[1]) + u_0
```

Perhaps you could come up with a better name for the expectation value of the cost operator.
I've used `_get_expect` just as a placeholder.

Reply:
@elisno Thanks for that, yeah, the number of steps definitely shouldn't be hardcoded.

On the other hand, I've different plans for the overall structure of loss calculation, but on the stage I'm on right now, this is good enough. Like I said, this is the initial contents, and is subject to heavy changes as I learn more :)